plugins/tiddlywiki/webdav/webdavadaptor.js

```diff
@@ -0,0 +1,248 @@
+/*\
+title: $:/plugins/tiddlywiki/webdav/webdavadaptor.js
+type: application/javascript
+module-type: syncadaptor
+
+A sync adaptor module for synchronising with WebDAV compatible servers
+Heavily based on WebDAV Adaptor
+
+Revised by Sebastian Silva <sebastian@fuentelibre.org>
+
+\*/
+(function(){
+
+/*jslint node: true, browser: true */
+/*global $tw: false */
+"use strict";
+
+var CONFIG_HOST_TIDDLER = "$:/config/webdav/host",
+	DEFAULT_HOST_TIDDLER = "$protocol$//$host$/";
+
+function dirname(path) {
+	return path.replace(/\\/g,'/').replace(/\/[^\/]*$/, '');
+}
+
+function WebDAVAdaptor(options) {
+	this.wiki = options.wiki;
+	this.host = this.getHost();
+	this.tiddlersBaseURL = dirname(location.pathname) + "/tiddlers";
+	this.hasStatus = false;
+	this.logger = new $tw.utils.Logger("WebDAVAdaptor");
+}
+
+WebDAVAdaptor.prototype.name = "webdav";
+WebDAVAdaptor.prototype.knownDirs = [];
+
+WebDAVAdaptor.prototype.isReady = function() {
+	return this.hasStatus;
+};
+
+WebDAVAdaptor.prototype.getHost = function() {
+	var text = this.wiki.getTiddlerText(CONFIG_HOST_TIDDLER,DEFAULT_HOST_TIDDLER),
```

Comment:
IMO this isn't a tiddler name, its a configuration value. right? I think DEFAULT_HOST should be ok.